test/cql-pytest/test_keyspace.py

```diff
@@ -191,6 +191,59 @@ def creates(count):
             cql.execute(f"DROP KEYSPACE {keyspace}")
         cql.execute(f"CREATE KEYSPACE {keyspace} {ksdef}")
 
+# Test that passing "S3" parameters to storage options works and is stored
+# in a schema table
+def test_storage_options_s3(cql, scylla_only):
+    ksdef = "WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : '1' } " \
+            "AND STORAGE = { 'type' : 'S3', 'bucket' : '42', 'endpoint' : 'localhost' }"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        res = cql.execute(f"SELECT * FROM system_schema.scylla_keyspaces WHERE keyspace_name = '{keyspace}'")
+        row = res.one()
+        assert row.storage_type == 'S3'
+        options = row.storage_options
+        assert options['bucket'] == '42'
+        assert options['endpoint'] == 'localhost'
+
+# Test that passing an unsupported storage type is not legal
+def test_storage_options_unknown_type(cql, scylla_only):
+    ksdef = "WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : '1' } " \
+            "AND STORAGE = { 'type' : 'S4', 'bucket' : '42', 'endpoint' : 'localhost' }"
+    with pytest.raises(InvalidRequest):
+        with new_test_keyspace(cql, ksdef):
+            pass
+
+# Test that passing nonexistent options results in an error
+def test_storage_options_nonexistent_param(cql, scylla_only):
+    ksdef = "WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : '1' } " \
+            "AND STORAGE = { 'type' : 'S3', 'bucket' : '42', 'endpoint' : 'localhost', 'superfluous' : 'info' }"
+    with pytest.raises(InvalidRequest):
+        with new_test_keyspace(cql, ksdef):
+            pass
+
+# Test that storage options cannot be altered (at least until it's well defined
+# what it means to e.g. switch from S3 to another format and back).
+def test_storage_options_alter_type(cql, scylla_only):
+    ksdef = "WITH REPLICATION = { 'class' : 'SimpleStrategy', 'replication_factor' : '1' } " \
+            "AND STORAGE = { 'type' : 'S3', 'bucket' : '42', 'endpoint' : 'localhost' }"
+    with new_test_keyspace(cql, ksdef) as keyspace:
+        # It's fine to amend if storage options aren't changed or just the parameters change
```

Comment:
I find it surprising that it's fine. How can it be fine to switch buckets in an existing table that already contains data? In what scenario will this make sense? If it makes sense to change the bucket, why doesn't it make sense to switch between local and s3?

Reply:
True, let's forbid it for now. If we ever find out that it's useful (for, I don't know, migrating to another location, another provider or whatnot), we can allow it back.